chore: Improves the core booking audit architecture by adding new capabilities and simplifying the existing interfaces.

hariombalhara opened the pull request:
## What does this PR do?

This PR improves the core booking audit architecture by adding new capabilities and simplifying the existing interfaces. This is a core-only extraction from PR #25125, containing no integration changes that could affect the booking flow.

### Key Enhancements:

**1\. New Action Services**

- `SeatBookedAuditActionService` - Tracks when a seat is booked in seated events
- `SeatRescheduledAuditActionService` - Tracks when a seat is rescheduled

**2\. Simplification of ActionService Interface**

- Refactored to use `GetDisplayTitleParams` and `GetDisplayJsonParams` interfaces for consistent stored data handling
- Reduces TypeScript burden by simplifying type inference across action services
- Removed generic type parameters from `IAuditActionService` interface

**3\. ActionSource Support**

- New `BookingAuditSource` enum (API_V1, API_V2, WEBAPP, WEBHOOK, UNKNOWN)
- Tracks where audit actions originate from
- Added `source` and `operationId` fields to BookingAudit model

**4\. New AuditAction Types**

- Added `SEAT_BOOKED` and `SEAT_RESCHEDULED` action types
- Added `APP` actor type in `AuditActorType` enum for payment webhook integrations

**5\. New BookingAuditAccessService**

- Permission-based access control for viewing audit logs
- Added `booking.readTeamAuditLogs` and `booking.readOrgAuditLogs` permissions
- Checks team/org permissions before allowing access

**6\. Fixes in the Logs Viewer Flow**

- Enhanced `BookingAuditViewerService` with improved filtering and display
    - Actor Filter uses Actor Name instead of Actor Type now
- Local `AttendeeRepository` for actor enrichment (internalized within booking-audit package)

**7\. Bulk Audit Task Support**

- New `BulkBookingAuditTaskConsumerSchema` for operations affecting multiple bookings
- Single `operationId` correlates related audit logs across bulk operations
- `BookingAuditTaskBasePayload` aliased to `BookingAuditTaskConsumerPayload` for backward compatibility

**Multiple System type actors supported** 

- Removed hardcoded general purpose system actor Id and introduced createSystemActorIfExists to create system actor as per requirement.

### New Repositories/DI 

| Repository | Location | Purpose |
| --- | --- | --- |
| `IAttendeeRepository` | `lib/repository/IAttendeeRepository.ts` | Interface for attendee data access |
| `PrismaAttendeeRepository` | `lib/repository/PrismaAttendeeRepository.ts` | Prisma implementation for attendee lookups |
| `AttendeeRepository.module.ts` | `di/AttendeeRepository.module.ts` | DI module for attendee repository |
| `AuditActorRepository.container.ts` | `di/AuditActorRepository.container.ts` | Container for audit actor repository access |

### Database Migration

Single migration (`20251216052900_add_booking_audit_core_enhancements`) containing all schema changes.

### Updates Since Last Revision

- **api-v2 Integration Fix**: Injected `auditActorRepository` dependency in `BookingAuditProducerService` via `getAuditActorRepository()`
- **Platform Libraries Export**: Added `getAuditActorRepository` export from `@calcom/platform-libraries/bookings`
- **Simplified Actor Handling**: `getPIIFreeBookingAuditActor` now only handles main branch actor types (user, guest, attendee, id)
- **Backward Compatible Tasker Payload**: `BookingAuditTaskBasePayload` now aliases to `BookingAuditTaskConsumerPayload` which supports both single and bulk operations

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a documentation change. N/A - internal infrastructure, documentation updated in [README.md](http://README.md) within the package.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. **Unit tests**: Run `TZ=UTC yarn test` - new tests in `BookingAuditAccessService.test.ts`, `BookingAuditViewerService.test.ts`, and `booking-audit.integration-test.ts` should pass
2. **Type checking**: Run `yarn type-check:ci --force` to verify no type errors introduced
3. **api-v2 build**: Run `cd apps/api/v2 && yarn build` to verify api-v2 compiles (note: 4 pre-existing errors on main will appear)
4. **Database migrations**: Verify migrations apply cleanly

## Human Review Checklist

- [ ] Verify permission registry changes (`ReadTeamAuditLogs`, `ReadOrgAuditLogs`) are correctly scoped
- [ ] Verify `PIIFreeActorSchema` in `lib/types/actor.ts` matches expected usage patterns
- [ ] **IMPORTANT**: Migration adds NOT NULL columns `source` and `operationId` - verify this handles existing data correctly or if defaults are needed
- [ ] Verify simplified `IAuditActionService` interface doesn't break existing action services
- [ ] Verify `BookingAuditTaskerProducerService` dependency on `auditActorRepository` is wired correctly in api-v2
- [ ] Verify `getAuditActorRepository()` container function works correctly at runtime

## Checklist

- [x] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings

---

**Link to Devin run:** https://app.devin.ai/sessions/5aa7cbaebf5e441d98c851977c3a4526
**Requested by:** @hariombalhara ([hariom@cal.com](mailto:hariom@cal.com))

devin-ai-integration[bot] committed:
feat(booking-audit): extract core audit system changes from PR 25125

This PR extracts core audit infrastructure changes without integration changes:

1. New action services introduced:
   - SeatBookedAuditActionService
   - SeatRescheduledAuditActionService

2. Simplification of ActionService interface:
   - Streamlined IAuditActionService interface
   - Reduced TypeScript burden with cleaner type definitions

3. ActionSource support:
   - Added BookingAuditSource enum (API_V1, API_V2, WEBAPP, WEBHOOK, UNKNOWN)
   - Added source and operationId fields to BookingAudit model

4. New AuditAction types:
   - SEAT_BOOKED
   - SEAT_RESCHEDULED
   - APP actor type

5. New BookingAuditAccessService:
   - Permission-based access control for audit logs
   - Added readTeamAuditLogs and readOrgAuditLogs permissions

6. Fixes in the logs viewer flow:
   - Enhanced BookingAuditViewerService with improved filtering
   - Local AttendeeRepository for actor enrichment

Changes are contained within packages/features/booking-audit with minimal
outside changes (permission registry only).

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>

hariombalhara commented on apps/web/modules/booking/logs/views/booking-logs-view.tsx:
```diff
@@ -51,9 +52,9 @@ type AuditLog = {
-    actorFilter: AuditActorType | null;
-    onActorFilterChange: (value: AuditActorType | null) => void;
-    actorOptions: Array<{ label: string; value: AuditActorType }>;
+    actorFilter: string | null;
```
Actor Filter was earlier based on Actor Type which was wrong. It should be based on Actor Name to easily identify who made what changes.

hariombalhara commented on apps/web/modules/booking/logs/views/booking-logs-view.tsx:
```diff
@@ -376,27 +421,6 @@ export default function BookingLogsView({ bookingUid }: BookingLogsViewProps) {
-    const uniqueActorTypes = Array.from(new Set(auditLogs.map((log) => log.actor.type))) as AuditActorType[];
-
-    const actorOptions = uniqueActorTypes.map((actorType) => ({
-        label: actorType,
```
Abstracted to hook and handled updated Actor Name based filter

hariombalhara commented on packages/features/bookings/lib/onBookingEvents/types.d.ts:
```diff
@@ -18,9 +18,11 @@ export interface BookingCreatedPayload {
+
 export interface BookingRescheduledPayload extends BookingCreatedPayload {
-  oldBooking?: {
+  oldBooking: {
```
old booking mus be there if it is being rescheduled

hariombalhara committed:
refactor(booking-audit): streamline action handling and enhance localization

- Replaced action icon retrieval with a mapping object for improved clarity and performance.
- Introduced constants for actor role labels to simplify role retrieval.
- Added new localization strings for audit log permission errors and organization requirements.
- Updated various service and repository interfaces to enhance type safety and clarity.
- Removed deprecated architecture documentation and adjusted related imports for consistency.

These changes aim to improve code maintainability and user experience in the booking audit system.

hariombalhara committed:
fix(booking-audit): enhance actor role localization and operation ID tracking

- Updated actor role labels in the booking logs view to use lowercase for consistency.
- Improved localization by wrapping actor role display in a translation function.
- Added operationId field to audit logs for better correlation of actions across multiple bookings.
- Enhanced BookingAuditViewerService to include operationId in enriched audit logs.
- Updated integration tests to verify consistent operationId across related audit logs.

These changes aim to improve localization accuracy and facilitate better tracking of user actions in the booking audit system.

joeauyeung commented on packages/features/booking-audit/lib/actions/ReassignmentAuditActionService.ts:
```diff
@@ -65,8 +64,8 @@ export class ReassignmentAuditActionService
-    async getDisplayTitle({ storedData }: { storedData: { version: number; fields: z.infer<typeof fieldsSchemaV1> }; userTimeZone: string }): Promise<TranslationWithParams> {
-        const { fields } = storedData;
+    async getDisplayTitle({ storedData }: GetDisplayTitleParams): Promise<TranslationWithParams> {
+        const { fields } = this.parseStored(storedData);
```
Curious why you made this change from passing `storedData` to method to storing it with the class.

I liked how the service was stateless so we could potentially resuse the single class for multiple audit actions. In this case multiple reassignment audit actions.

joeauyeung reviewed (dismissed):
Great PR and seeing how the booking audit is shaping. I know this is still a WIP so my comments are non-blocking and can always be followed up in a separate PR if needed.

hariombalhara commented on packages/features/bookings/lib/types/actor.ts:
```diff
@@ -92,5 +99,20 @@ export function makeAttendeeActor(attendeeId: number): AttendeeActor {
+ * Creates an Actor representing an App by app slug
+ * App actors use @app.internal email convention
+ */
+export function makeAppActor(params: { appSlug: string; name: string }): AppActor {
```
Yeah absolutely makes sense!! Infact required. Will tackle it in followup  
  
As actor doesn't have app specific fields. I think we could have appSlug as well as credential together in email in some format, will think about it

hariombalhara replied:
Resolved it

hariombalhara committed:
feat: integrate credential repository and enhance app actor handling

- Added CredentialRepository to manage app credentials, including a method to find credentials by ID.
- Updated BookingAudit system to support app actors identified by credential ID, improving actor attribution and audit clarity.
- Introduced a new utility function to map app slugs to display names, enhancing the user experience in audit logs.
- Modified relevant interfaces and types to accommodate the new credential handling and app actor structure.
- Enhanced BookingAuditViewerService to display app names based on credentials, ensuring accurate representation in audit logs.

joeauyeung reviewed (approved):
LGTM and my comments are resolved. Looking forward to the follow ups
